Assignment2_q1.py

```diff
@@ -0,0 +1,80 @@
+# coding=utf-8
+from BinaryTree import TreeNode, BinaryTree
+
+"""Prints all the ancestors of the key in the given binary tree. Uses two methods:
+   first method 'print_ancestors' takes a tree and a key and calls the second helper
+   recursive method 'print_ancestors_recursive' which also takes tree and a key but it
+   also passes a list of ancestors so far as an argument.
+   Given solution takes θ(n) time, where n is the number of nodes in the tree."""
+
+
+def print_ancestors(tree, key):
+    """Prints all the ancestors of the key in the given binary tree and returns a list of them.
+       Ancestors contain the key itself. Ancestors are ordered: first in the list (and first printed)
+       is the key itself then it 'crawls up' the tree up until the root.
+
+        Args:
+            tree: a binary tree, tree to check for ancestors, no duplicated keys allowed
+            key: a key type in a given tree, a key for the ancestors to be printed
+
+        Returns:
+            a list of key type, list of the ancestors of the key and prints ancestors
+
+        Throws:
```

Comment:
I think it is better to use `Raises:` instead of `Throws:` (apologies if I have actually suggested the wrong one before). Because Python syntax `raise`s exceptions.